source/common/stats/symbol_table_impl.h

```diff
@@ -400,6 +407,52 @@ class StatNameManagedStorage : public StatNameStorage {
   SymbolTable& symbol_table_;
 };
 
+/**
+ * Maintains storage for a collection of StatName objects. Like
+ * StatNameManagedStorage, this has an RAII usage model, taking
+ * care of decrementing ref-counts in the SymbolTable for all
+ * contained StatNames on destruction or on clear();
+ *
+ * Example usage:
+ *   StatNamePool pool(symbol_table);
+ *   StatName name1 = pool.add("name1");
+ *   StatName name2 = pool.add("name2");
+ *   uint8_t* storage = pool.addReturningStorage("name3");
+ *   StatName name3(storage);
+ */
+class StatNamePool {
+public:
+  explicit StatNamePool(SymbolTable& symbol_table) : symbol_table_(symbol_table) {}
+  ~StatNamePool() { clear(); }
+
+  /**
+   * Removes all StatNames from the pool.
+   */
+  void clear();
+
+  /**
+   * @param name the name to add the container.
+   * @return the StatName held in the container for this name.
+   */
+  StatName add(absl::string_view name);
+
+  /**
+   * Does essentially the same thing as add(), but returns the storage as a
+   * pointer which can later be used to create a StatName. This can be used
+   * to accumulate a vector of uint8_t* which can later be used to create
+   * StatName objects on demand.
```

Comment:
I'm a little confused by this. When would we do this?

Reply:
This is used in #6733 but maybe it's best to remove it here and add it in that PR; you may have better ideas when that PR is ready about how to accomplish the same thing.

It's used  there to provide a lock-free way of finding a StatName associated with a cluster and an HTTP response code, only taking a lock if one had not already been created for a specific response code. Would you prefer I move it to #6733 and discuss there? It is at least tested here.

Reply:
Yeah it would be better to just discuss there I think if you don't mind.

Reply:
done